tests/console/zypper_lifecycle.pm

```diff
@@ -42,22 +42,21 @@ our $date_re = qr/[0-9]{4}-[0-9]{2}-[0-9]{2}/;
 sub run {
     diag('fate#320597: Introduce \'zypper lifecycle\' to provide information about life cycle of individual products and packages');
 
-    # Workaround for bsc#1166549, zypper lifecycle error with 'Error building the cache'.
-    # Currently the only workaround option is to run zypper-lifecycle as root.
-
-    # Now we just hit bsc#1166549 on s390x and ppc64le, so we just need root console
-    # on these two platforms.
-    my $needs_root_console = 0;
-    if ((is_s390x || is_ppc64le) && is_sle('>=15-SP3') && is_upgrade) {
-        $needs_root_console = 1;
-        record_soft_failure 'bsc#1166549 - zypper lifecycle failed to create metadata cache directory';
-    }
     select_console 'root-console';
-
+    # first we'd make sure that zypper cache dir have 0755 permission
+    # for some reason the system will change the permission on /var/cache/zypp/
+    # this cause the zypper lifecycle failed while building the cache
+    my $fmode = script_output("ls -l /var/cache/zypp/ | grep solv awk -F \" \" \'{print \$1}\'");
+    if ($fmode ne 'drwxr-xr-x') {
```

Comment:
I think the condition is not correct, it should be at least 0755 while not must be 0755 since it maybe 0777 or others which do not need to do chmod.

Reply:
Do you mean >=0755?

Reply:
changed to use stat not ls, please take a look again.